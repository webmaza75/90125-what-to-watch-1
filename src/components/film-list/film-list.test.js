import React from 'react';
import renderer from 'react-test-renderer';

import FilmList from './film-list.jsx';

window.HTMLMediaElement.prototype.play = () => {};

const mock = {
  genre: `Drama`,
  title: `Macbeth`,
  picture: `macbeth.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 1
};

it(`Film list correctly renders`, () => {
  const tree = renderer
    .create(<FilmList
      films={[mock]}
      onChange={jest.fn()}
      activeItem={mock}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
