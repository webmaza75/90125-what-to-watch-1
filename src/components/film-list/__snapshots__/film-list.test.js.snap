// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film list correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <a
      href="/film/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="War of the Worlds"
          height="175"
          src="https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        War of the Worlds
      </a>
    </h3>
  </article>
</div>
`;
